webapp/javascript/redux/actions.js

```diff
@@ -395,3 +426,67 @@ export function abortFetchNames() {
     }
   };
 }
+
+let profilesController;
+export function fetchProfiles() {
+  return (dispatch) => {
+    if (profilesController) {
+      profilesController.abort();
+    }
+
+    profilesController = new AbortController();
+    dispatch(requestProfiles());
+    return fetch('/adhoc/v1/profiles', {
+      signal: profilesController.signal,
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        dispatch(receiveProfiles(data));
+      })
+      .catch((e) => {
+        if (!isAbortError(e)) {
```

Comment:
Let's dispatch an error notification here!

Reply:
I think this can be done more easily once this PR is merged https://github.com/pyroscope-io/pyroscope/pull/625

Reply:
Right, once that's merged I'll change these accordingly :+1:

Reply:
I ended up overlapping the functionality with changes in  https://github.com/pyroscope-io/pyroscope/pull/624/commits/75bcb8f9e7caaa87b3c41b42d9e45b7a77251ecd and https://github.com/pyroscope-io/pyroscope/pull/624/commits/53188ec9436ec0c9e304d967a8158a56702db8a1 :facepalm:

Reply:
I combined the approaches when merging at https://github.com/pyroscope-io/pyroscope/pull/624/commits/f9d0fdac3555d9f77c2fbeb00173b8fcade0bbef
Also added cancellation support in https://github.com/pyroscope-io/pyroscope/pull/624/commits/2471f56976b5b41ad087d3d5aa7021a3155d9f46